pkg/aqua/command.go

```diff
@@ -68,61 +66,46 @@ func (c *command) Exec(imageRef ImageRef) (report ScanReport, err error) {
 		image = imageRef.WithTag()
 	}
 
-	flags := []string{
-		"--user", c.cfg.Username,
-		"--password", c.cfg.Password,
-		"--host", c.cfg.Host,
-		"--registry", c.cfg.Registry,
+	args := []string{
+		"scan",
+		"--checkonly",
 		"--dockerless",
-		"--jsonfile", reportFile.Name(),
-	}
-
-	if c.cfg.ScannerCLINoVerify {
-		flags = append(flags, "--no-verify")
-	}
-
-	if c.cfg.ScannerCLIShowNegligible {
-		flags = append(flags, "--show-negligible")
+		fmt.Sprintf("--user=%s", c.cfg.Username),
+		fmt.Sprintf("--password=%s", c.cfg.Password),
+		fmt.Sprintf("--host=%s", c.cfg.Host),
+		fmt.Sprintf("--registry=%s", c.cfg.Registry),
+		fmt.Sprintf("--no-verify=%t", c.cfg.ScannerCLINoVerify),
+		fmt.Sprintf("--show-negligible=%t", c.cfg.ScannerCLIShowNegligible),
+		fmt.Sprintf("--show-will-not-fix=%t", c.cfg.ScannerCLIShowWillNotFix),
+		fmt.Sprintf("--hide-base=%t", c.cfg.ScannerCLIHideBase),
+		fmt.Sprintf("--jsonfile=%s", reportFile.Name()),
+		image,
 	}
 
-	args := append([]string{"scan"}, flags...)
-	args = append(args, image)
-
-	log.WithFields(log.Fields{"exec": executable, "args": args}).Trace("Running scannercli")
+	// TODO Sanitize the args so the password is not printed no matter what's the log level!
```

Comment:
I think this is worth doing before release. You could populate everything except the password and image, log that, and then append the image and password

Reply:
Agree. I fixed it:

```
{"args":["scan","--checkonly","--dockerless","--user=administrator","--host=http://csp-console-svc.aqua:8080","--registry=Harbor","--no-verify=false","--show-negligible=true","--show-will-not-fix=false","--hide-base=true","--jsonfile=/var/lib/scanner/reports/aqua_scan_report_679963226.json"],"exec":"/usr/local/bin/scannercli","level":"debug","msg":"Running scannercli","time":"2020-04-22T11:56:35Z"}
```